src/main/java/com/mcmiddleearth/guidebook/data/InfoArea.java

```diff
@@ -69,7 +76,9 @@ public abstract class InfoArea {
 
     private List<String> description = new ArrayList<>();
 
-    protected InfoArea() {
+    protected InfoArea(String areaName) {

```

Comment:
Since the GuidebookSendEvent is called inside `InfoArea` the area needed to know its own name